app/src/main/java/lt/vilnius/tvarkau/SignInActivity.java

```diff
@@ -0,0 +1,104 @@
+package lt.vilnius.tvarkau;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.view.View;
+import android.widget.Toast;
+
+import com.google.android.gms.auth.api.Auth;
+import com.google.android.gms.auth.api.signin.GoogleSignInAccount;
+import com.google.android.gms.auth.api.signin.GoogleSignInOptions;
+import com.google.android.gms.auth.api.signin.GoogleSignInResult;
+import com.google.android.gms.common.ConnectionResult;
+import com.google.android.gms.common.api.GoogleApiClient;
+
+import butterknife.ButterKnife;
+import butterknife.OnClick;
+
+
+/*
+    Activity which handles log in of user.
+    It does not matter whether user decides to do it,
+    he will be transferred to MainActivity
+
+ */
+public class SignInActivity extends AppCompatActivity implements GoogleApiClient.OnConnectionFailedListener{
```

Comment:
I have been been thinking do we really need separate activity for login ? In home screen we have sign in button. After clicking this button user will get login dialog with account selection. I think having separate `SignInActivity` is unnecessary and also not very very convenient for user (he needs one more click to just select if he wants to sign in or not). 
![screenshot_20160123-112633](https://cloud.githubusercontent.com/assets/3719141/12529583/ebbe6640-c1c4-11e5-88e8-9039acb78f4d.png)